### PR TITLE
[FIX] Scheduler Issues by updating Agenda

### LINKED ISSUE #1158: [Bug][Priority] RC sometimes not sending zero timer payloads?
I noticed that some sessions on the brazil bot were getting no response at times from the DF.agent.

https://ba.chatbot.vega.viasat.com/live/sBkcCy9aeq3ujxdpq/room-info
https://ba.chatbot.vega.viasat.com/live/QBdfwBdq657aza5mC/room-info
https://ba.chatbot.vega.viasat.com/live/gtP8QotNBXLEf746e/room-info

Molly suggested that DF was expecting a zero timer payload which never arrived from RC.

```
hey, so i am digging into the first session you sent, and https://ba.chatbot.vega.viasat.com/live/sBkcCy9aeq3ujxdpq is getting stuck on a particular page in CX that expects a RC zero timer event - but RC didn’t send the event
8:35
not sure why as that flow has been workign fine for me every time i’ve tested
8:37
is there any way to see why RC is not sending that event? it seems like a weird intermittent issue?
8:38
```

@Shailesh351 I will assign to you......  maybe there is something in the RC logs?

### PR DESCRIPTION
**Pull request by Shailesh351**
Closes: https://github.com/WideChat/Rocket.Chat/issues/1158

Widechat Agenda Fork: https://github.com/WideChat/agenda
Agenda Version: https://github.com/WideChat/agenda/releases/tag/v4.3.1

@ear-dev All the features dependent on Scheduling API need to be tested (i.e. Timer Payload, Visitor Abandonment Timer, Session Maintenance ...)

### CODE REVIEW

File: app/apps/server/bridges/scheduler.ts [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
+import { Agenda, Job } from 'agenda';
 import { ObjectID } from 'bson';
 import { MongoInternals } from 'meteor/mongo';
 import { StartupType, IProcessor, IOnetimeSchedule, IRecurringSchedule } from '@rocket.chat/apps-engine/definition/scheduler';
 import { SchedulerBridge } from '@rocket.chat/apps-engine/server/bridges/SchedulerBridge';
 
 import { AppServerOrchestrator } from '../orchestrator';
 
-function _callProcessor(processor: Function): (job: Agenda.Job) => void {
+const { MONGO_URL = 'mongodb://localhost:3001/meteor' } = process.env;
```
Comment (ear-dev) [inline app/apps/server/bridges/scheduler.ts]: Is the mongo URL actually on port 27017? Or is that /meteor telling it something else?

Comment (Shailesh351) [reply app/apps/server/bridges/scheduler.ts]: You are right. On my local it was `3001/meteor` but it should be 27017. I'm updating it. Thanks!

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -8,11 +8,12 @@ on:
   push:
     branches:
       - develop_pwa
+  workflow_dispatch:
 
 env:
   CI: true
   MONGO_URL: mongodb://localhost:27017
-  TOOL_NODE_FLAGS: --max_old_space_size=4096
+  TOOL_NODE_FLAGS: --max_old_space_size=8192 --optimize_for_size --gc-interval=100
```
Comment (ear-dev) [inline .github/workflows/build_and_test.yml]: Do we think this helped?  As I recall it did not originally help and that our issue just seemed to fix itself?

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -29,6 +30,11 @@ jobs:
           echo "github.event_name: ${{ github.event_name }}"
           cat $GITHUB_EVENT_PATH
 
+      - name: Set Swap Space
```
Comment (ear-dev) [inline .github/workflows/build_and_test.yml]: Do we think this helped?  Do we want to keep it?

File: .github/workflows/build_and_test.yml [hunk 4]
```diff
@@ -132,7 +138,7 @@ jobs:
         env:
           METEOR_PROFILE: 1000
         run: |
-          meteor build --server-only --directory --debug /tmp/build-test
+          meteor build --server-only --directory --debug /tmp/build-test --verbose
```
Comment (ear-dev) [inline .github/workflows/build_and_test.yml]: Did we learn anything using "--verbose"?  Do we want to keep it?

Comment (Shailesh351) [reply .github/workflows/build_and_test.yml]: Nothing at all. It logged only downloading dependencies and extracting them. Nothing useful.

File: .github/workflows/build_and_test.yml (hunk 3 above)
Comment (Shailesh351) [reply .github/workflows/build_and_test.yml]: Yeah, I think adding swap space worked!. I'm going to clean up this PR and create separate PR for fixing build issue

File: .github/workflows/build_and_test.yml (hunk 2 above)
Comment (Shailesh351) [reply .github/workflows/build_and_test.yml]: Nope It didn't work

**Review by ear-dev — APPROVED** [review]

**Pull request merged**